resources/api/reservation.py

```diff
@@ -267,13 +304,21 @@ def perform_create(self, serializer):
             instance.send_created_by_admin_mail()
 
     def perform_update(self, serializer):
+        kwargs = {'modified_by': self.request.user}
         old_instance = self.get_object()
-        new_instance = serializer.save(modified_by=self.request.user)
+        new_state = serializer.validated_data.get('state', old_instance.state)
+        if new_state == Reservation.CONFIRMED and old_instance.state != Reservation.CONFIRMED:
```

Comment:
Hmm, but then it doesn't work the same way? That is meant to deal only with changes to and from `CONFIRMED`